account_invoice_report_grouped_by_picking_hide_pickings/models/account_move.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def lines_grouped_by_picking(self):
+        res = super().lines_grouped_by_picking()
+        to_remove = []
+        for group in res:
+            if group['picking'].hide_on_invoice:
```

Comment:
esto podría ponerse como filter/filtered en el for anterior y así ya vas borrando directamente mientras recorres los que tocan

Reply:
No es un recordset, no esta la función filtered

Reply:
pero si la de filter que es nativa de python